src/refactorings/change-signature/change-signature.test.ts

```diff
@@ -0,0 +1,404 @@
+import { Code, ErrorReason, RelativePath } from "../../editor/editor";
+import { InMemoryEditor } from "../../editor/adapters/in-memory-editor";
+import { testEach } from "../../tests-helpers";
+
+import { changeSignature } from "./change-signature";
+
+type TestSample = {
+  code: Code;
+  path: RelativePath;
+};
+
+describe("Change Signature", () => {
```

Comment:
👍 Kudos for adding tests that document the different scenarios this is already handling. Love it!